jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildConfiguration.java

```diff
@@ -201,6 +203,7 @@ public BuildConfiguration build() {
     @VisibleForTesting
     ImmutableList<String> expandPortRanges(List<String> ports) throws NumberFormatException {
       ImmutableList.Builder<String> result = new ImmutableList.Builder<>();
+      Pattern portPattern = Pattern.compile("(\\d+)(-\\d+|)(\\/tcp|\\/udp|)");
```

Comment:
I just tested it more and realized you can just do 

```
(\d+)(?:-(\d+))?(/tcp|/udp)?
```

And it'll match those groups, with them being null if it didn't match.

Reply:
I don't see any advantage to doing this, just seems harder to understand and getting null instead of an empty string just forces us to add more checks.